std/string.d

```diff
@@ -5798,36 +5798,57 @@ unittest
  * function, or any of the conversion functions.
  *
  * Params:
- *     s = the string to check
+ *     s = the string or random access range to check
  *     bAllowSep = accept separator characters or not
  *
  * Returns:
  *     $(D bool)
  */
-bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
+bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S ||
+    (isRandomAccessRange!S &&
+    hasSlicing!S &&
+    isSomeChar!(ElementEncodingType!S) &&
+    !isInfinite!S))
```

Comment:
nitpick: why don't you add `@safe @nogc pure nothrow` here?
I thought a recent policy is too explicitly state it, s.t. it's guaranteed for all types and future versions of the method.